forc-test/src/lib.rs

```diff
@@ -396,6 +397,14 @@ impl<'a> PackageTests {
                 .entries
                 .par_iter()
                 .filter_map(|entry| entry.kind.test().map(|test| (entry, test)))
+                .filter(|(entry, _)| {
+                    // If a test filter is specified, only the tests containing the filter phrase in
+                    // their name are going to be executed.
+                    match &test_filter {
+                        Some(filter) => entry.finalized.fn_name.contains(filter),
+                        None => true,
+                    }
+                })
```

Comment:
We should unit test this functionality:
- No filter
- Filter matches nothing
- Filter matches something
- Exact match filter

Reply:
Good idea, added tests for:

1. No filter
2. Exact filter matches nothing
3. Exact filter matching a thing
4. Non exact filter matches nothing
5. Non exact filter fetches a single thing
6. Non exact filter matches everything